### DISCUSSION TITLE
Unit Testing for BC Functions

### DISCUSSION THREAD

**Discussion by jtneedels** [su2code/SU2#1881] [Q&A]
It seems like unit testing would be a good way to verify BC behavior, rather than just through regression testing. For BC functions with no return type, is there a good way to write a unit test?

**Comment by pcarruscag** [comment]
The only thing I can think of is calling the BC and checking the residuals agains expected values.

**Reply by jtneedels** [reply]
Got it, so creating dummy classes to pass in then checking then running and checking the residual that gets output?

**Reply by pcarruscag** [reply]
Have a look at the UnitQuad unit tests
